src/back-end/lib/crud.ts

```diff
@@ -86,3 +87,37 @@ export function makeRouter<SupportedRequestBodies, SupportedResponseBodies, Crea
     return routes;
   };
 }
+
+interface BodyWithErrors {
+  permissions?: string[];
+  database?: string[];
+}
+
+interface ResponseValidation<ValidBody, InvalidBody, ResB, Session> {
+  valid: Respond<ValidBody, ResB, Session>;
+  invalid: Respond<InvalidBody, ResB, Session>;
+}
+
+export function wrapRespond<ValidBody, InvalidBody extends BodyWithErrors, ResB, Session>(responseValidation: ResponseValidation<ValidBody, InvalidBody, ResB, Session>): Respond<Validation<ValidBody, InvalidBody>, ResB, Session> {
```

Comment:
@dhruvio can you confirm this is correct? Had a bit of trouble getting the types working for the inner `respond` function. Wouldn't mind doing a quick review with you before using through the CRUD resources.

Reply:
Had a quick look, and it looks great!